span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanPreProcessor.java

```diff
@@ -91,14 +94,27 @@ PreProcessedSpan preProcessSpan(Span span) {
         span.getProcess().getTagsList().stream()
             .collect(Collectors.toMap(t -> t.getKey().toLowerCase(), t -> t, (v1, v2) -> v2));
 
-    if (spanDropManager.shouldDropSpan(span, spanTags, processTags)) {
+    Optional<String> tenantIdMaybe =
+        tenantIdHandler.getAllowedTenantId(span, spanTags, processTags);
+    if (tenantIdMaybe.isEmpty()) {
+      return null;
+    }
+    String tenantId = tenantIdMaybe.get();
+    Event event =
+        buildEvent(tenantId, span, tenantIdHandler.getTenantIdProvider().getTenantIdTagKey());
```

Comment:
Can we store `tenantIdHandler.getTenantIdProvider().getTenantIdTagKey()` in a separate variable?